src/portlet/Portlet.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import {classNames} from "../utils";
+import {Icon, Direction} from "../icon";
+
+type PortletProps = {
+    style?: any;
+    className?: string;
+    header: any;
+};
+
+type PortletState = {
+    toggleContent: boolean;
+    iconStyle: any;
+};
+
+export class Portlet extends React.PureComponent<PortletProps, PortletState> {
+    state: PortletState = {
+        toggleContent: true,
+        iconStyle: {
+            transition: "all .3s ease-in 0s",
+            color: "#007DB8",
+        },
+    };
+    handleClick = () => {
+        const {toggleContent} = this.state;
+        this.setState({toggleContent: !toggleContent});
+        toggleContent
```

Comment:
I think it is better to use if, since you don't need the result of this.setState and you just process the logic

Reply:
addressed